calvalus-processing/src/main/java/com/bc/calvalus/processing/l2/CombineFileInputFormat.java

```diff
@@ -0,0 +1,83 @@
+package com.bc.calvalus.processing.l2;
+
+import com.bc.calvalus.JobClientsMap;
+import com.bc.calvalus.commons.CalvalusLogger;
+import com.bc.calvalus.commons.InputPathResolver;
+import com.bc.calvalus.inventory.hadoop.FileSystemPathIterator;
+import com.bc.calvalus.inventory.hadoop.HdfsFileSystemService;
+import com.bc.calvalus.processing.JobConfigNames;
+import com.bc.calvalus.processing.geodb.GeodbScanMapper;
+import com.bc.calvalus.processing.hadoop.NoRecordReader;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.CombineFileSplit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author thomas
+ */
+public class CombineFileInputFormat extends InputFormat {
+
+    /**
+     * Creates a single split from a given pattern
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext context) throws IOException {
```

Comment:
What about our other methods to determine inputs, in particular those using the geo-inventory? I know that PatternBasedInputFormat needs refactoring and decomposition but I think the other ways to determine inputs are required.
Thinking of how to refactor PatternBasedInputFormat it may be good to distinguish the way the inputs shall be determined (geo-inventory, opensearch query, path pattern, ...) by different classes as they have different parameters anyway, and whichever parameter is specified the client could automatically select the right class. Then, we could either derive a class for CombineFileSplit generation from each of them, or we make this a parameter. In any case, the old PatternBasedInputFormat could delegate the getSplits() call to the new implementations to keep backwards compatibility.